### ISSUE TITLE
Extending AuthProviders for auth version 1.0?

### ISSUE THREAD

**Issue by kengorab** [fullstackreact/react-native-oauth#169]
Judging from the conversation in #64 it appears that the customization functionality could only be used to add new 2.0 providers. Indeed, looking at the [`getApiFor10aProvider` method in `OAuthManagerProviders`](https://github.com/fullstackreact/react-native-oauth/blob/master/android/src/main/java/io/fullstack/oauth/OAuthManagerProviders.java#L42), it's not possible to add custom 1.0 providers.

At least, not for Android. As best I can tell from reading over the [`OAuth1Client.m`](https://github.com/fullstackreact/react-native-oauth/blob/master/ios/OAuthManager/OAuth1Client.m#L58) code, any generic oauth 1.0 provider can be configured for iOS.

It appears as though it should just involve adding another method which mimics the [`OAuth10aService twitterService(...)`](https://github.com/fullstackreact/react-native-oauth/blob/master/android/src/main/java/io/fullstack/oauth/OAuthManagerProviders.java#L159) method (as well as testing, etc). Granted, I've only been reading through this code for the past 25 minutes or so.

Is it on the roadmap for this project to support custom oauth 1.0 providers for Android? I'd be willing to help implement it. Also, please let me know if I've gotten anything wrong in this issue, too.

Thanks!

**Comment by corradio** [comment]
@kengorab did you get any news? Are you using an alternative implementation?

**Comment by kengorab** [comment]
Yeah, I ended up rolling my own, unfortunately. Sorry, I know that's not very helpful to you

**Comment by corradio** [comment]
I did a fork here that supports it partially https://github.com/tmrowco/react-native-oauth

**Comment by kg-currenxie** [comment]
@corradio any intentions of finishing your fork? Just by pure luck, I'm also implementing TripIt :D 
I noticed `// TODO: GET THE OAUTH CALLBACK` :(
I could fork..

**Comment by corradio** [comment]
Hi @kg-currenxie,

The fork actually works for TripIt. However, we ended re-implementing everything in javascript and handling the Linking with RCTLinking.
We've open sourced part of the code here:
https://github.com/tmrowco/tmrowapp-contrib/blob/master/integrations/utils/oauth.js

**Comment by kg-currenxie** [comment]
Haha, so did I, also using oauth-1.0a, but got stuck at signing step 3. Thanks for your link, ill check it out :)
